drivers/adc/adc_b91.c

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2022 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+#include "adc.h"
+#include <zephyr/dt-bindings/adc/b91-adc.h>
+
+#define LOG_LEVEL CONFIG_ADC_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(adc_b91);
+
+#define DT_DRV_COMPAT telink_b91_adc
```

Comment:
`"adc.h"` is not a local header, but from HAL I guess. Please improve this include section, you can find some hints here: https://github.com/zephyrproject-rtos/zephyr/issues/41543